### ISSUE TITLE
DOC: OptimizeResult Notes are placed before attribute section, but refers to them as if placed after

### ISSUE THREAD

**Issue by FirefoxMetzger** [scipy/scipy#14683]
This is a minor issue, and I wonder if it isn't a bug in either numpydoc or sphinx.

The docstring of `OptimizeResult` correctly contains the `Notes` section at the bottom and states `There may be additional attributes not listed above[...]`. However, in the rendered docs the `Notes` section is actually placed _before_ the other sections making the reference a bit strange (the attributes are listed below).

Here is the the link to the rendered docs: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.OptimizeResult.html#scipy.optimize.OptimizeResult

**Comment by mdhaber** [comment]
Thanks @FirefoxMetzger. PR welcome. For anyone interested, please see the [contributor guide](http://scipy.github.io/devdocs/dev/contributor/contributor_toc.html#scipy-contributor-guide).

**Comment by FirefoxMetzger** [comment]
@mdhaber What kind of fix would be preferred? Updating the notes to be generic and say `additional attributes not listed here` or to change the position of `Notes`?

**Comment by mdhaber** [comment]
In the [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html#parameters), "Parameters" is section #4 and "Attributes" of a class is supposed to go below "Parameters". "Notes" is toward the bottom at #13, so I suppose that changing the order would be best.

**Comment by FirefoxMetzger** [comment]
@mdhaber In that case I'm not sure how easy this fix will be. In the source code already does the correct thing and lists "Notes" after "Attributes":

https://github.com/scipy/scipy/blob/47bb6febaa10658c72962b9615d5d5aa2513fa3a/scipy/optimize/optimize.py#L83-L118

which means there is something not going as intended during the doc building process; no idea where things break.

**Comment by mdhaber** [comment]
OK, in that case, just the generic "additional attributes not listed here" is fine.

**Comment by FirefoxMetzger** [comment]
For future reference, I just checked the order of sections in a repo I maintain that uses a very similar setup (sphinx+pydata-theme+numpydoc) and found that there, too, the `Notes` section is placed above the `Attributes` section when documenting classes: https://scikit-bot.org/en/latest/_autosummary/skbot.transform.Link.html#skbot.transform.Link

I threw out the pydata-theme in my local dev environment (using sphinx default `alabaster` instead), and the resulting docs still place the `Notes` section above `Attributes`.

In conclusion, this appears to be either a feature or a bug of `numpydoc`; nothing scipy can do much about. I'll see if I can find a related issue upstream and if not I'll see when I can find the time to make a MWE and ask upstream.

**Comment by FirefoxMetzger** [comment]
Looks like this has been recently raised upstream: https://github.com/numpy/numpydoc/issues/331
